cpp/src/ray/config_internal.cc

```diff
@@ -25,6 +25,14 @@
 
 ABSL_FLAG(std::string, ray_address, "", "The address of the Ray cluster to connect to.");
 
+/// absl::flags does not provide a IsDefaultValue method, so use a non-empty dummy default
+/// value to support empty redis username.
+ABSL_FLAG(std::string,
+          ray_redis_username,
+          "absl::flags dummy default value",
```

Comment:
can we use empty string as default value?

Reply:
The reason for using a dummy default value is to allow overriding the value with an empty string, see https://github.com/ray-project/ray/pull/17633/files#r685874199.

In other words, if we use an empty string as the default value, we will have the following:

```c
if (!FLAGS_ray_redis_username.CurrentValue().empty()) {
  redis_username = FLAGS_ray_redis_username.CurrentValue();
}
```
instead of the current one:
```c
if (FLAGS_ray_redis_username.CurrentValue() !=
    FLAGS_ray_redis_username.DefaultValue()) {
  redis_username = FLAGS_ray_redis_username.CurrentValue();
}
```
The first case will not be able to override the `redis_username` with a `--ray_redis_username=""` flag. I think we better keep it handled in the same way as the `redis_password`.

Reply:
today I learned redis can have empty username and empty password...

Reply:
Yes, redis can even have a non-empty username without a password. In that case, a user needs to specify a non-empty password in order to make the `AUTH` command syntax work.

<img width="902" alt="image" src="https://github.com/user-attachments/assets/e6d49b48-bc2f-44ce-9838-2cb8250bce30">
https://redis.io/docs/latest/operate/oss_and_stack/management/security/acl/